# HG changeset patch
# User Rik <rik@octave.org>
# Date 1479488555 28800
#      Fri Nov 18 09:02:35 2016 -0800
# Node ID 92e7def948c2e53e6bf219ef1aef63d0c8ffc121
# Parent  d90bc9cdc638b11f950bfa02c5388f9369218abb
Present questdlg buttons in correct order (bug #49644).

* dialog.cc (MessageDialog::MessageDialog): Reverse for loop that adds buttons
to go from low to high.

diff --git a/libgui/src/dialog.cc b/libgui/src/dialog.cc
--- a/libgui/src/dialog.cc
+++ b/libgui/src/dialog.cc
@@ -152,17 +152,17 @@ MessageDialog::MessageDialog (const QStr
     eicon = QMessageBox::Question;
   setIcon (eicon);
 
   int N = qsbutton.size () < role.size () ? qsbutton.size () : role.size ();
   if (N == 0)
     addButton (QMessageBox::Ok);
   else
     {
-      for (int i = N-1; i >= 0; i--)
+      for (int i = 0; i < N; i++)
         {
           // Interpret the button role string, because enumeration
           // QMessageBox::ButtonRole can't be made to pass through a signal.
           QString srole = role.at (i);
           QMessageBox::ButtonRole erole = QMessageBox::InvalidRole;
           if (srole == "YesRole")
             erole = QMessageBox::YesRole;
           else if (srole == "NoRole")
